Compiler/src/Builtins.cpp

```diff
@@ -166,6 +168,11 @@ static int getBuiltinFunctionId(const Builtin& builtin, const CompileOptions& op
             return LBF_BIT32_COUNTLZ;
         if (builtin.method == "countrz")
             return LBF_BIT32_COUNTRZ;
+        if (builtin.method == "byteswap")
+        {
+            LUAU_ASSERT(FFlag::LuauBit32ByteswapBuiltin);
+            return LBF_BIT32_BYTESWAP;
+        }
```

Comment:
The way to approach flagging the code is to make sure that without the flag enabled, code runs as if new parts were not there.
Right now, is someone has `bit32.byteswap` call in their code for some reason, they will start hitting the assertion failure when flag is disabled.
The correct approach here is to check the flag in the `if` statement:
`if (FFlag::LuauBit32ByteswapBuiltin && builtin.method == "byteswap")`